src/Spout/Writer/XLSX/Writer.php

```diff
@@ -125,7 +125,7 @@ protected function getDefaultRowStyle()
      */
     protected function closeWriter()
     {
-        if ($this->book) {
+        if ($this->book && $this->isWriterOpened) {
```

Comment:
This addition should be moved to the `close()` function inside `AbstractWriter` so it's applied to all writers and not only the XLSX one.